### ISSUE TITLE
Determine calls for Release Administrator App backend

### ISSUE THREAD

**Issue by milen-sage** [Sage-Bionetworks/release_administrator#3]
When app UI is ready

- [ ] outline what information is needed from the backend to populate the current UI elements
- [ ] confirm that corresponding  REST API endpoints exist in schematic and/or Synapse and note the ones that are missing
- [ ] test one or more REST API call(s) retrieve the accurte information in the right format for the app

Acceptance criteria: at least one UI element derives real data using the REST API.

**Comment by lakikowolfe** [comment]
### `mod_dataset_selection`

1. List all storage projects a user has access to

*API functions from [DCA](https://github.com/Sage-Bionetworks/data_curator/blob/schematic-rest-api/R/schematic_rest_api.R)*
```
storage_projects <- function(url="http://localhost:3001/v1/storage/projects",
                             asset_view,
                             input_token) {
  
  req <- httr::GET(url,
                   query = list(
                     asset_view=asset_view,
                     input_token=input_token
                   ))
  
  httr::content(req)
}

storage_projects_list <- storage_projects(asset_view = "syn20446927",
                                          input_token = input_token)
```
Example output:
```
[[1]]
[[1]][[1]]
[1] "syn20834712"

[[1]][[2]]
[1] "HTAN HTAPP"


[[2]]
[[2]][[1]]
[1] "syn21050481"

[[2]][[2]]
[1] "HTAN Vanderbilt"


[[3]]
[[3]][[1]]
[1] "syn22041595"

[[3]][[2]]
[1] "HTAN TNP - TMA"
```
Feed this to a drop down like in DCA? Or will the user want to see across all storage projects?

2. For each project or each project selected pull down associated datasets

```
storage_project_datasets <- function(url="http://localhost:3001/v1/storage/project/datasets",
                                     asset_view,
                                     project_id,
                                     input_token) {
  
  req <- httr::GET(url,
                   #add_headers(Authorization=paste0("Bearer ", pat)),
                   query=list(
                     asset_view=asset_view,
                     project_id=project_id,
                     input_token=input_token)
  )
  
  httr::content(req)
}

### EX OUTPUT FOR HTAN HTAP

[[1]]
[[1]][[1]]
[1] "syn23585912"

[[1]][[2]]
[1] "merfish_level_1"


[[2]]
[[2]][[1]]
[1] "syn23585913"

[[2]][[2]]
[1] "merfish_level_2"


[[3]]
[[3]][[1]]
[1] "syn23585914"

[[3]][[2]]
[1] "merfish_level_3"
```
Open Qs: How do you know which of the listed datasets have manifests? Some seem empty, but I'll need to investigate this more.

3. User selects datasets, for selected datasets pull manifests

I don't think there is an API endpoint for this yet. It sounds like initially manifests will be downloaded as csv and filepath will be output. Possibly could get json output from endpoint.

4. User selects files to update status for, corresponding manifests are updated with new status, timestamp, etc

5. Updated manifest is pushed to synapse. I think we can use `/model/submit/` for this.

**Comment by lakikowolfe** [comment]
>I don't think there is an API endpoint for this yet. It sounds like initially manifests will be downloaded as csv and filepath will be output. Possibly could get json output from endpoint.

[csv download+filepath completed](https://github.com/Sage-Bionetworks/schematic/issues/632)

**Comment by ychae** [comment]
@lakikowolfe would it make sense to keep this open for sprint 6 to wrap up in the shortened sprint?

**Comment by lakikowolfe** [comment]
yes! I think this will be the case for both of my tickets this week.

**Comment by lakikowolfe** [comment]
**R function for manifest download**

manifest_download <- function(url="http://localhost:3001/v1/manifest/download",
                              asset_view,
                              dataset_id,
                              input_token) {
  # define headers
  headers = c(
    `accept` = 'text/csv'
  )
  
  # define params
  params = list(
    `input_token` = input_token,
    `asset_view` = asset_view,
    `dataset_id` = dataset_id
  )
  
  # make request
  req <- httr::GET(url = url,
                   httr::add_headers(.headers=headers), 
                   query = params)
  
  # parse request
  httr::content(req, as="text")
}

**Will use model/submit to upload modified manifest**
